lib/Connection.php

```diff
@@ -249,7 +256,7 @@ private function closeSocket() {
         $this->outputBuffer = "";
         $this->outputBufferLength = 0;
 
-        if (is_resource($this->socket)) {
+        if (!$this->persistent && is_resource($this->socket)) {
```

Comment:
Could you separate the timeout fix into its own commit and PR from the other changes?